src/compliment/utils.clj

```diff
@@ -181,36 +185,37 @@ Note that should always have the same value, regardless of OS."
       (->> (for [^String file (all-files-on-classpath classpath)
                  :when (and (.endsWith file ".class") (not (.contains file "__"))
                             (not (.contains file "$")))]
-             (.. (if (.startsWith file File/separator)
-                   (.substring file 1) file)
+             (.. (ensure-no-leading-slash file)
                  (replace ".class" "")
                  ;; Address the issue #79 , on Windows, for prefix such
                  ;; as "java.util.", the list of candidates was empty.
                  (replace resource-separator ".")))
            (group-by #(subs % 0 (max (.indexOf ^String % ".") 0)))))))
 
-(defn namespaces-on-classpath
-  "Returns the list of all Clojure namespaces obtained by classpath scanning."
+(defn namespaces&files-on-classpath
+  "Returns a collection of maps (e.g. `{:ns-str \"some.ns\", :file \"some/ns.cljs\"}`) of all clj/cljc/cljs namespaces obtained by classpath scanning."
   []
   (let [classpath (classpath)]
-    (cache-last-result ::namespaces-on-classpath classpath
-      (set (for [^String file (all-files-on-classpath classpath)
-                 :when (and (.endsWith file ".clj")
-                            (not (.startsWith file "META-INF")))
-                 :let [[_ ^String nsname] (re-matches #"[^\w]?(.+)\.clj" file)]
-                 :when nsname]
-             (.. nsname (replace resource-separator ".") (replace "_" "-")))))))
+    (set (cache-last-result ::namespaces-on-classpath classpath
```

Comment:
`set` moved out of the caching form which is bad, we want to cache that transformation too. However, in the new function, `set` doesn't achieve what it used to do before for strings, since it is unlikely that we'll have duplicate filenames (but there could be duplicate namespaces). So, to match the prior behavior, we need something like `distinct-by` which does not exist. Or might just leave it as is for now (by removing `set`) and maybe place a `;; TODO` for future deduplication.

Reply:
Good point - added the TODO, but also wondering if it's relevant now that we return clj, cljc and cljs results?

Reply:
I think this is quite minor, to be honest. There is a slim chance that two same namespaces on the classpath come from different files, and somebody using this new function will show both of them (but not Compliment, since it performs its own deduplication later). And that is... not a big issue:).